File: leetcode2/1easy/최원준/Q3477.java
package Leetcode;

/*
1. 아이디어 :


2. 시간복잡도 :
O( )

3. 자료구조/알고리즘 :

 */

import java.util.HashSet;
import java.util.Set;

public class Q3477 {
    class Solution {
        public int numOfUnplacedFruits(int[] fruits, int[] baskets) {
            int n = fruits.length;
            Set<Integer> visited = new HashSet<>();
            for (int fruit : fruits) for (int i = 0; i<n; i++) {
                if (!visited.contains(i) && baskets[i] >= fruit) {
                    visited.add(i);
                    break;
                }
            }
            return n - visited.size();
        }
    }
}

File: leetcode2/1easy/최은서/Q3477.js
/**
 * @param {number[]} fruits
 * @param {number[]} baskets
 * @return {number}
 */
var numOfUnplacedFruits = function(fruits, baskets) {
    //fruits를 하나씩 baskets의 왼쪽부터 돌면서 못 넣은 과일 개수 리턴
    let res = 0;
    const n = fruits.length;
    for (let i = 0; i < n; i++) {
        let flag = false;
        for (let j = 0; j < n; j++) {
            if (fruits[i] <= baskets[j]) {
                baskets[j] = -1;
                flag = true
                break;
            }
        }
        if (!flag) {
            res += 1;
        }
    }
    return res
};